data_model/primitives/src/lib.rs

```diff
@@ -15,3 +15,91 @@ extern crate alloc;
 pub mod atomic;
 pub mod fixed;
 pub mod small;
+
+use fixed::prelude::*;
+
+/// Trait for values that could be converted to `f64` and observed by prometheus.
+pub trait FloatMetricFrom {
+    /// Convert `value` to `f64`.
+    fn from(value: Self) -> f64;
+}
+
+impl FloatMetricFrom for u128 {
+    #[inline]
+    #[allow(clippy::cast_precision_loss)]
+    fn from(value: Self) -> f64 {
+        value as f64
+    }
+}
+
+impl FloatMetricFrom for u32 {
+    #[inline]
+    fn from(value: Self) -> f64 {
+        value.into()
+    }
+}
+
+impl FloatMetricFrom for Fixed {
+    #[inline]
+    fn from(value: Self) -> f64 {
+        value.into()
+    }
+}
+
+/// Trait for `checked_add` operation on iroha's primitive types.
+pub trait CheckedAdd {
+    /// Checked addition. Computes `self + rhs`, returning None if overflow occurred.
+    fn checked_add(self, rhs: Self) -> Option<Self>
+    where
+        Self: Sized;
+}
+
+impl CheckedAdd for u32 {
+    #[inline]
+    fn checked_add(self, rhs: Self) -> Option<Self> {
+        self.checked_add(rhs)
+    }
```

Comment:
This works because `Self::checked_add()` has bigger priority than `<Self as CheckedAdd>::checked_add()`, yes?
This trick is also used in `tools/parity_scale_decoder/src/generate_map.rs`

Reply:
Yes, rust book has example of such behavior, [see](https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name).